### ISSUE TITLE
RedundantReturn: false negative in lambda

### ISSUE THREAD

**Issue by rnveach** [sevntu-checkstyle/sevntu.checkstyle#700]
Talked about in https://github.com/sevntu-checkstyle/sevntu.checkstyle/pull/686

````
$ cat TestClass.java
public class TestClass {
    public interface Something {
        void action();
    }

    public static void main(String args[]) {
        Something l = () -> {
            System.out.println("Do nothing!");
            return;
        };
        l.action();
    }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<module name="com.github.sevntu.checkstyle.checks.coding.RedundantReturnCheck"/>
    </module>
</module>

$ java -jar checkstyle-8.9-sevntu-1.29.0-all.jar -c TestConfig.xml TestClass.java
Starting audit...
Audit done.
````

Expecting a violation on line 9.
